src/query_error.c

```diff
@@ -95,10 +112,18 @@ void QueryError_MaybeSetCode(QueryError *status, QueryErrorCode code) {
   status->code = code;
 }
 
-const char *QueryError_GetError(const QueryError *status) {
+const char *QueryError_GetUserError(const QueryError *status) {
   return status->detail ? status->detail : QueryError_Strerror(status->code);
 }
 
+const char *QueryError_GetDisplayableError(const QueryError *status, bool obfuscate) {
+  if (status->detail == NULL || obfuscate) {
```

Comment:
Sure we can't have `detail = NULL` even when we don't want obfuscation?
Sounds ok since these are static error messages thus they should be in `message`, but please validate that we get the expected behavior when setting the `code` only (i.e., `detail` is `NULL`, regardless of the obfuscation requirement that may not exist.

Reply:
In `QueryError_SetCode` we don't set detail or message and only set the code.
But that seems to keep in line with the behaviour we want.

Reply:
Something didn't add up for me but at a second look it's fine 👌